File: be-todo-app/models/task.go
package models

import (
	"time"
)

type Task struct {
	ID          uint       `gorm:"primaryKey" json:"id"`
	Name        string     `gorm:"not null" json:"name"`
	IsComplete  bool       `gorm:"default:false" json:"is_complete"`
	CompletedAt *time.Time `json:"completed_at"`
	CreatedAt   time.Time  `json:"created_at"`
}

File: fe-todo-app/src/services/task.ts
import api from "./api";

export interface ITask {
  id: number;
  name: string;
  is_complete: boolean;
  completed_at?: string;
  created_at?: string;
}

export const getTodoList = () => {
  return api.get<{ todoList: ITask[] }>("/tasks");
};

export const createTask = (name: string) => {
  return api.post<{ task: ITask }>("/tasks", { name });
};

export const updateTask = (id: number, task : ITask) => {
  return api.put<{ task: ITask }>(`/tasks/${id}`, task);
};

export const deleteTask = (id: number) => {
  return api.delete(`/tasks/${id}`);
};
